components/org.wso2.carbon.event.simulator.core/src/main/java/org/wso2/carbon/event/simulator/core/service/SimulationConfigDeployer.java

```diff
@@ -34,39 +34,69 @@
     private void deployConfigFile(File file) throws Exception {
         if (!file.getName().startsWith(".")) {
             if (FilenameUtils.isExtension(file.getName(), EventSimulatorConstants.SIMULATION_FILE_EXTENSION)) {
                 String simulationName = FilenameUtils.getBaseName(file.getName());
-                String simulationConfig = SimulationConfigUploader.getConfigUploader().getSimulationConfig
-                        (simulationName, (Paths.get(Utils.getCarbonHome().toString(),
-                                EventSimulatorConstants.DIRECTORY_DEPLOYMENT,
-                                EventSimulatorConstants.DIRECTORY_SIMULATION_CONFIGS)).toString());
-                if (!simulationConfig.isEmpty()) {
-                    EventSimulator simulator = new EventSimulator(simulationName, simulationConfig);
-                    EventSimulationMap.getSimulatorMap().put(simulationName, simulator);
+                try {
+                    String simulationConfig = SimulationConfigUploader.getConfigUploader().getSimulationConfig
+                            (simulationName, (Paths.get(Utils.getCarbonHome().toString(),
+                                    EventSimulatorConstants.DIRECTORY_DEPLOYMENT,
+                                    EventSimulatorConstants.DIRECTORY_SIMULATION_CONFIGS)).toString());
+                    if (!simulationConfig.isEmpty()) {
+                        EventSimulator eventSimulator = new EventSimulator(simulationName, simulationConfig);
+                        EventSimulatorMap.getInstance().getActiveSimulatorMap().put(simulationName,
+                                Collections.singletonMap(eventSimulator, simulationConfig));
+                        log.info("Successfully deployed active simulation '" + simulationName + "'.");
+                    }
+                } catch (ResourceNotFoundException e) {
+                    EventSimulatorMap eventSimulatorMap = EventSimulatorMap.getInstance();
+                    if (eventSimulatorMap.containsInActiveSimulator(simulationName)) {
+                        if (!eventSimulatorMap.getResourceTypeForInActiveSimulator(simulationName)
+                                .equals(e.getResourceType())
+                                || !eventSimulatorMap.getResourceNameForInActiveSimulator(simulationName)
+                                .equals(e.getResourceName())) {
+                            eventSimulatorMap.deleteInActiveSimulation(simulationName);
+                            eventSimulatorMap.getInActiveSimulatorMap().put(simulationName, Collections
+                                    .singletonMap(e.getResourceType(), e.getResourceName()));
+                        }
+                    } else {
+                        eventSimulatorMap.getInActiveSimulatorMap().put(simulationName, Collections
+                                .singletonMap(e.getResourceType(), e.getResourceName()));
+                        log.error(e.getMessage(), e);
```

Comment:
You should not say "Successfully deployed inactive simulation, " say "moved simulation to inactive state".